lib/unison-prelude/src/Unison/Prelude.hs

```diff
@@ -59,9 +63,17 @@ import qualified System.IO as IO
 import Text.Read as X (readMaybe)
 import qualified UnliftIO
 
-onNothing :: Applicative m => m a -> Maybe a -> m a
-onNothing x =
-  maybe x pure
+-- | E.g.
+--
+-- @@
+-- onNothing (throwIO MissingPerson) $ mayThing
+-- @@
+onNothing :: Applicative m => Maybe a -> m a -> m a
```

Comment:
Hm, I think the old `onNothing` was better, especially with the addition of the new `whenNothing`. Why was it changed?

To me, the `on*` naming scheme works for functions that take the scrutinee first (following Control.Monad.when`:

```haskell
when :: Bool -> m () -> m ()
whenNothing :: Maybe a -> m a -> m a
```

etc.

That frees up the `on*` naming convention for the flipped variants, which are useful in postfix style, e.g.

```haskell
thing <- whatever & onLeft \theLeft -> do
  stuffWithTheLeft
...
```